resources/js/contact.js

```diff
@@ -0,0 +1,292 @@
+// https://youtu.be/eg4e-FObyJ8
+const submitBtn = document.querySelector(".formBtn")
+const fields = document.querySelectorAll(".input");
+const form = document.querySelector('form')
+
+var isValid = true
+
+submitBtn.addEventListener('click', function(){
+  validateFields(fields)
+  
+  if (isValid) {
+  
+    var contactData = {
+      name: fields[0].value,
+      email: fields[3].value,
+      phone: fields[2].value,
+      message: fields[4].value,
+    };
+    
+    const contactJSON = JSON.stringify(contactData);
+    // clearError()
+    apiCall(contactJSON);
+  } else {
+    isValid = true;
+  }
+  
+})
+
+function validateFields(fields){
+  
+  for(i=0; i<fields.length;i++){
+    var input = fields[i]
+    validate(input)
+  }
+}
+
+function validate(input) {
+  var validName = false;
+  var validEmail = false;
+  var validPhone = false;
+  var validMessage = false;
+  if (input.required && input.value.length == 0) {
+    setError(input, `${input.name} field cannot be empty!`);
+  } else {
+    if (input.id == "name") {
+      validName = validateName(input)
+    }
+    else if(input.id == "phone"){
+      validPhone = validatePhone(input)
+    }
+    else if(input.id == "email"){
+      validEmail = validateEmail(input)
+    }
+    else if(input.id == "message"){
+      validMessage = validateMessage(input)
+    }
+  }
+
+}
+
+function validateName(input){
+  if(input.value.length < 8){
+    setError(input, "Name is too short!");
+    return false
+  }else{
+    if(!isName(input.value)){
+      setError(input, "Name is invalid!");
+      return false
+    }
+    else{
+      clearError(input)
+      return true
+    }
+  }
+}
+
+function validatePhone(input){
+  
+  if(isNaN(input.value) || input.value.length != 10){
+    setError(input, "Phone number is invalid!");
+    return false;
+  }else{
+    clearError(input)
+    return true
+  }
+}
+
+function validateEmail(input){
+  if (!isEmail(input.value)) {
+    setError(input,"Email is invalid!");
+    return false
+  } else {
+    clearError(input);
+    return true
+  }
+}
+
+function validateMessage(input){
+  if (input.value.length < 6) {
+    setError(input, "Message is too short!");
+    return false
+  } else {
+    clearError(input);
+    return true
+  }
+}
+
+function countChar(input){
+  const counter = document.querySelector(`#${input.id}-counter`);
```

Comment:
@Thejus5 Could have stored this counter value in a data-attribute in the input elm itself, rather than using a separate HTML element for this.